wire.go

```diff
@@ -132,3 +133,22 @@ func Value(interface{}) ProvidedValue {
 func InterfaceValue(typ interface{}, x interface{}) ProvidedValue {
 	return ProvidedValue{}
 }
+
+// StructFields is a collection of the fields from a struct.
+type StructFields struct{}
+
+// FieldsOf extracts the structType into a list of types specified by fieldNames.
```

Comment:
I'd like to reword this a bit for clarity. Perhaps:

FieldsOf declares that the fields named of the given struct type will be used to provide the types of those fields.

Reply:
Done.